openslides_backend/shared/mixins/user_scope_mixin.py

```diff
@@ -179,134 +179,193 @@ def check_permissions_for_scope(
         Also checks requested user has no committee management rights if a dict was provided instead of an id.
+        An ID will be provided during user deletion.
         Returns true if permissions are given. False if not. Raises no Exceptions.
         """
-        if isinstance(instance_or_id, dict):
-            if not (instance_id := instance_or_id.get("id")):
+        if not self._check_not_committee_manager(instance_id):
+            return False
+
+        if not (
+            intersection_meetings := self._collect_intersected_meetings(b_meeting_ids)
+        ):
+            return False
+        assert isinstance(intersection_meetings, dict)
+        admin_meeting_users = self._collect_admin_meeting_users(intersection_meetings)
+        meeting_to_admin_users = self._collect_admin_users(admin_meeting_users)
+        return self._analyze_meetings(meeting_to_admin_users, instance_id)
+
+    def _check_not_committee_manager(self, instance_id: int) -> bool:
+        """
+        Helper function used in method check_for_admin_in_all_meetings.
+        Checks that requested user is not a committee manager.
+        """
+        if not (hasattr(self, "name") and self.name == "user.create"):
+            if self.datastore.get(
+                fqid_from_collection_and_id("user", instance_id),
+                ["committee_management_ids"],
+                lock_result=False,
+                use_changed_models=False,
+            ).get("committee_management_ids", []):
                 return False
-            if hasattr(self, "name") and self.name == "user.create":
-                if instance_or_id.get("committee_management_ids", []):
-                    return False
-            else:
-                user = self.datastore.get(
-                    fqid_from_collection_and_id("user", instance_id),
-                    ["committee_management_ids"],
-                    lock_result=False,
-                    use_changed_models=False,
-                )
-                if user.get("committee_management_ids", []):
-                    return False
-        else:
-            instance_id = instance_or_id
+        return True
+
+    def _collect_intersected_meetings(
+        self, b_meeting_ids: set[int] | None
+    ) -> dict[int, Any] | bool:
+        """
+        Helper function used in method check_for_admin_in_all_meetings.
+        Takes the meeting ids to find intersections with the requesting users meetings. Returns False if this is not possible.
+        """
         if not b_meeting_ids:
-            if not hasattr(self, "instance_committee_meeting_ids"):
+            if not (
+                b_meeting_ids := {
+                    m_id
+                    for m_ids in self.instance_committee_meeting_ids.values()
+                    for m_id in m_ids
+                }
+            ):
                 return False
```

Comment:
Can this be shortened, there are two if clauses nested inside each other with no further content on the outer one

Reply:
Yes it can be handled with 'and'. The outer one is there for performance reasons. There is no need to calculate b_meeting_ids if they are given. If the calculation also delivers nothing, False should be returned.